sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/AesCbcDecryptOptions.java

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.security.keyvault.keys.cryptography;
+
+/**
+ * A class containing configuration parameters that can be applied when decrypting AES-CBC keys with and without
+ * padding.
+ */
+public class AesCbcDecryptOptions extends DecryptOptions {
+    /**
+     * Creates an instance of {@link AesCbcDecryptOptions} with the given parameters.
+     *
+     * @param iv Initialization vector for the decryption operation.
+     */
+    public AesCbcDecryptOptions(byte[] iv) {
```

Comment:
The Key size is specified in the clients. For example: `CryptographyClient.encrypt(EncryptionAlgorithm algorithm, String plaintext, EncryptOptions options)`.

Reply:
Do you have any thoughts on this @srnagar?

Reply:
Since these are inputs to the API, how does the user know what are the available subtypes of EncryptOptions/DecryptOptions. In other places, we have used a type flag to switch between strongly typed sub-classes.
See example here:
https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FieldValue.java